DataManagementSystem/Agent/RequestOperations/MoveReplica.py

```diff
@@ -7,201 +7,324 @@
-    gMonitor.registerActivity( "RemoveReplicaFail", "Failed replica removals",
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
+    super(MoveReplica, self).__init__(operation, csPath)
 
     # Init ConsistencyInspector: used to check replicas
     self.ci = ConsistencyInspector()
 
-  def __call__( self ):
+  def __call__(self):
     """ call me maybe """
+
+    # Check whether the ES flag is enabled so we can send the data accordingly.
+    if self.rmsMonitoring:
+      self.rmsMonitoringReporter = MonitoringReporter(monitoringType="RMSMonitoring")
+    else:
+      # # own gMonitor stuff for files
+      gMonitor.registerActivity("ReplicateAndRegisterAtt", "Replicate and register attempted",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("ReplicateOK", "Replications successful",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("ReplicateFail", "Replications failed",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RegisterOK", "Registrations successful",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RegisterFail", "Registrations failed",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RemoveReplicaAtt", "Replica removals attempted",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RemoveReplicaOK", "Successful replica removals",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RemoveReplicaFail", "Failed replica removals",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+
     # # check replicas first
     res = self.__checkReplicas()
     if not res["OK"]:
-      self.log.error( 'Failed to check replicas', res["Message"] )
+      self.log.error('Failed to check replicas', res["Message"])
 
     sourceSE = self.operation.SourceSE if self.operation.SourceSE else None
     if sourceSE:
       # # check source se for read
-      bannedSource = self.checkSEsRSS( sourceSE, 'ReadAccess' )
+      bannedSource = self.checkSEsRSS(sourceSE, 'ReadAccess')
       if not bannedSource["OK"]:
-        gMonitor.addMark( "ReplicateAndRegisterAtt", len( self.operation ) )
-        gMonitor.addMark( "ReplicateFail", len( self.operation ) )
+        if self.rmsMonitoring:
+          for opFile in self.operation:
+            for status in ["FileAttempted", "FileFailed"]:
+              self.rmsMonitoringReporter.addRecord({
+                  "timestamp": time.mktime(datetime.datetime.utcnow().timetuple()),
```

Comment:
According to the various docs (`time` and `datetime`, this does not give utc)

Reply:
I had verified this and it gives timestamp in accordance with UTC time.

Reply:
I am not sure 
https://docs.python.org/2/library/datetime.html#datetime.datetime.timetuple
https://docs.python.org/2/library/datetime.html#datetime.datetime.utctimetuple
https://docs.python.org/2/library/time.html#time.gmtime


```
In [9]: map(int, (time.mktime(datetime.datetime.utcnow().utctimetuple()), time.mktime(datetime.datetime.utcnow().timetuple()), time.mktime(datetime.datetime.now().timetuple()), time.mktime(d
   ...: atetime.datetime.now().utctimetuple()),  time.mktime(time.gmtime())))
Out[9]: [1566199416, 1566195816, 1566203016, 1566206616, 1566199416]
```

Reply:
the trick is often the day light saving factor

Reply:
Maybe better to use the time method from DIRAC?
```
from DIRAC.Core.Utilities import Time
Time.toEpoch()
```
or please check how it is done in that module...

Reply:
Okay will check this.

Reply:
@chaen @zmathe I checked this `Time.toEpoch()` it uses this `datetime.datetime(2000, 1, 1).utcnow().timetuple()` and I think in our case this will fit `datetime.datetime(2000, 1, 1).utcnow().utctimetuple()` because in this case `tm_isdst=0` so it confirms that the time is in `UTC` otherwise if we use `timetuple` then `tm_isdst=-1` so it is assumed that it is an unkown factor.
So should we go with this `datetime.datetime(2000, 1, 1).utcnow().utctimetuple()`?

Reply:
Hum, tricky. But Zoltan is right, use `Time.toEpoch()`. maybe it is not correct (maybe it is), but at least it is consistent everywhere

Reply:
Okay fine I will go with `Time.toEpoch()`.